websites/H/Hitomi/presence.ts

```diff
@@ -0,0 +1,123 @@
+    "/index-romanian.html": "romanian",
+    "/index-serbian.html": "serbian",
+    "/index-albanian.html": "albanian",
+    "/index-slovak.html": "slovak",
+    "/index-finnish.html": "finnish",
+    "/index-swedish.html": "swedish",
+    "/index-tagalog.html": "tagalog",
+    "/index-vietnamese.html": "vietnamese",
+    "/index-turkish.html": "turkish",
+    "/index-greek.html": "greek",
+    "/index-bulgarian.html": "bulgarian",
+    "/index-mongolian.html": "mongolian",
+    "/index-russian.html": "russian",
+    "/index-ukrainian.html": "ukrainian",
+    "/index-hebrew.html": "hebrew",
+    "/index-arabic.html": "arabic",
+    "/index-persian.html": "persian",
+    "/index-thai.html": "thai",
+    "/index-korean.html": "korean",
+    "/index-chinese.html": "chinese",
+    "/index-japanese.html": "japanese"
+  },
+  // /(type)/(title)-(lang)-(number).html
+  validateInfoUrl = /\/(.+)\/(.+)-(.+)-(\d+).html/,
+  // (number)
+  validateReaderUrl = /\/reader\/(\d+).html/;
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "app"
+  };
+
+  presenceData.details = "Viewing recently added list";
+  presenceData.state = "Home";
+
+  if (document.location.pathname in pathMapping) {
+    presenceData.details = "Viewing recently added list";
+    presenceData.state = pathMapping[document.location.pathname];
+  }
+
+  if (validateInfoUrl.exec(document.location.pathname)) {
+    const parsedUrl = validateInfoUrl.exec(document.location.pathname),
+      type = hitomiTypeMapping[parsedUrl[1]];
+    let title = document.querySelector(
+      `body > div > div.content > div.gallery.${type}-gallery > h1 > a`
+    ).textContent;
+
+    if (title.length > 128) title = `${title.slice(0, 120)}...`;
+
+    presenceData.details = title;
+    presenceData.state = `${
+      document.querySelector(
+        `body > div > div.content > div.gallery.${type}-gallery > h2 > ul > li > a`
+      ).textContent
+    } (${parsedUrl[4]})`;
+    presenceData.buttons = [
+      { label: "View Page", url: document.location.href }
+    ];
+  }
+
+  if (validateReaderUrl.exec(document.location.pathname)) {
+    let title = document.title.replace(" | Hitomi.la", "");
+
+    if (title.length > 128) title = `${title.slice(0, 120)}...`;
+
+    const selectValue = document.querySelector(
+        "#single-page-select"
+      ) as HTMLSelectElement,
+      totalPage = selectValue.options[selectValue.options.length - 1].value;
+    presenceData.details = title;
+    presenceData.state = `Reading page ${document.location.hash.replace(
+      "#",
+      ""
+    )} of ${totalPage} (${
+      validateReaderUrl.exec(document.location.pathname)[1]
+    })`;
+    presenceData.buttons = [
+      { label: "View Page", url: document.location.href }
+    ];
+  }
+
+  if (presenceData.details == null) {
+    presence.setTrayTitle();
+    presence.setActivity();
+  } else {
+    presence.setActivity(presenceData);
+  }
```

Comment:
Unnecessary brace for one-line code.

Reply:
The coding style should be decided by the maintainer.

https://github.com/PreMiD/Presences/blob/master/websites/R/RLSwaps/presence.ts#L30-L34
https://github.com/PreMiD/Presences/blob/master/websites/Y/YouTube/presence.ts#L576-L581

...more

Please take another look at the other code before reviewing or pointing it out.

Reply:
Yeah... I mean you could do that. Every new presence should have good code quality. Older ones just don't have it.

Reply:
I'm going to update waking rules today so yeah.

Reply:
It would be best to follow the code style of the project, but I think it should unified with the same code style in at least one file.

Base on @SaidBySolo's opinion, these should have changes too.
https://github.com/PreMiD/Presences/blob/f459026aef24bf0e2b9afa11b8390587cf3ad3ad/websites/H/Hitomi/presence.ts#L83
https://github.com/PreMiD/Presences/blob/f459026aef24bf0e2b9afa11b8390587cf3ad3ad/websites/H/Hitomi/presence.ts#L99

Reply:
@wonderlandpark

I'm new to TypeScript and it's the first time I first know that parentheses can be omitted.

I just respected the maintainer's opinion.

It's uncomfortable to see that a third party I see for the first time raises these issues.

Reply:
https://github.com/PreMiD/Presences/commit/d07a62ed697a988ef25dcef7b3586440e9c639a5 just fixed it and please don't argue

Reply:
Still not resolved